sky/clouds/service_catalog/data_fetchers/fetch_gcp.py

```diff
@@ -431,12 +431,16 @@ def get_vm_price(row: pd.Series, spot: bool) -> float:
         if series in ['f1', 'g1']:
             memory_price = 0.0
 
-        assert cpu_price is not None, row
-        assert memory_price is not None, row
+        # TODO(tian): (2024/11/10) Some SKUs are missing in the SKUs API. We
+        # skip them in the catalog for now. We should investigate why they are
+        # missing and add them back.
+        if cpu_price is None or memory_price is None:
+            return None
         return cpu_price + memory_price
 
     df['Price'] = df.apply(lambda row: get_vm_price(row, spot=False), axis=1)
     df['SpotPrice'] = df.apply(lambda row: get_vm_price(row, spot=True), axis=1)
+    df = df.dropna(subset=['Price', 'SpotPrice'])
```

Comment:
Can we print out those ones with NA, so we can keep a track of what VMs are missing those info?

Reply:
Done! Seems like only n2 series is missing now.

```bash
Price not found for n2-highcpu-16, n2-highcpu-2, n2-highcpu-32, n2-highcpu-4, n2-highcpu-48, n2-highcpu-64, n2-highcpu-8, n2-highcpu-80, n2-highcpu-96, n2-highmem-128, n2-highmem-16, n2-highmem-2, n2-highmem-32, n2-highmem-4, n2-highmem-48, n2-highmem-64, n2-highmem-8, n2-highmem-80, n2-highmem-96, n2-standard-128, n2-standard-16, n2-standard-2, n2-standard-32, n2-standard-4, n2-standard-48, n2-standard-64, n2-standard-8, n2-standard-80, n2-standard-96. Dropping them.
```

Reply:
Hmm, are we dropping all the rows that does not have spot OR on-demand price? What if the VMs that only has on-demand?

Reply:
Good point! Updated. atm it seems the two are identical, but agree that it will helps future proof :)

```bash
Price not found for n2-highcpu-16, n2-highcpu-2, n2-highcpu-32, n2-highcpu-4, n2-highcpu-48, n2-highcpu-64, n2-highcpu-8, n2-highcpu-80, n2-highcpu-96, n2-highmem-128, n2-highmem-16, n2-highmem-2, n2-highmem-32, n2-highmem-4, n2-highmem-48, n2-highmem-64, n2-highmem-8, n2-highmem-80, n2-highmem-96, n2-standard-128, n2-standard-16, n2-standard-2, n2-standard-32, n2-standard-4, n2-standard-48, n2-standard-64, n2-standard-8, n2-standard-80, n2-standard-96. Dropping them.
```

Reply:
Oh, I am a bit confused by the output. I assume we are only dropping those instances within a single region or zone? `n2-highcpu` is heavily used in many cases.

Reply:
It seems like the pricing info is only missed in the region `northamerica-south1`.

```bash
Price not found for the following zones and instance types. Dropping them.
------------------------------ northamerica-south1-a ------------------------------
n2-highcpu-16, n2-highcpu-2, n2-highcpu-32, n2-highcpu-4, n2-highcpu-48, n2-highcpu-64, n2-highcpu-8, n2-highcpu-80, n2-highcpu-96, n2-highmem-128, n2-highmem-16, n2-highmem-2, n2-highmem-32, n2-highmem-4, n2-highmem-48, n2-highmem-64, n2-highmem-8, n2-highmem-80, n2-highmem-96, n2-standard-128, n2-standard-16, n2-standard-2, n2-standard-32, n2-standard-4, n2-standard-48, n2-standard-64, n2-standard-8, n2-standard-80, n2-standard-96
------------------------------ northamerica-south1-b ------------------------------
n2-highcpu-16, n2-highcpu-2, n2-highcpu-32, n2-highcpu-4, n2-highcpu-48, n2-highcpu-64, n2-highcpu-8, n2-highcpu-80, n2-highcpu-96, n2-highmem-128, n2-highmem-16, n2-highmem-2, n2-highmem-32, n2-highmem-4, n2-highmem-48, n2-highmem-64, n2-highmem-8, n2-highmem-80, n2-highmem-96, n2-standard-128, n2-standard-16, n2-standard-2, n2-standard-32, n2-standard-4, n2-standard-48, n2-standard-64, n2-standard-8, n2-standard-80, n2-standard-96
------------------------------ northamerica-south1-c ------------------------------
n2-highcpu-16, n2-highcpu-2, n2-highcpu-32, n2-highcpu-4, n2-highcpu-48, n2-highcpu-64, n2-highcpu-8, n2-highcpu-80, n2-highcpu-96, n2-highmem-128, n2-highmem-16, n2-highmem-2, n2-highmem-32, n2-highmem-4, n2-highmem-48, n2-highmem-64, n2-highmem-8, n2-highmem-80, n2-highmem-96, n2-standard-128, n2-standard-16, n2-standard-2, n2-standard-32, n2-standard-4, n2-standard-48, n2-standard-64, n2-standard-8, n2-standard-80, n2-standard-96
```